.github/workflows/benchmarks.yml

```diff
@@ -26,6 +41,9 @@ jobs:
       GOBENCH_TAGS: branch=${{ github.head_ref || github.ref }},commit=${{ github.sha }},target_branch=${{ github.base_ref }}
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.inputs.ref }}
```

Comment:
What is the reason to explicitly allow any ref to be checkout?

Reply:
To be able to run benchmarks on any branch or forked PR optionally when desired

Reply:
The workflow dispatch form already should already provide input to choose the branch.
As of today, there isn't a clear way to support securely forked PRs.
I'd prefer to avoid an explicit checkout for now until we settle on the forked PRs support.

Reply:
https://github.com/elastic/apm-server/pull/11119/commits/cf99bdb0f717d9305196c5910383c80b2cd6a02d